### DISCUSSION TITLE
Scenario of distributed agent

### DISCUSSION THREAD

**Discussion by vincent-pli** [microsoft/autogen#3998] [Q&A] [answered]
That's great to see the distributed agents implementation, that's excited from technique perspective: the agent can run anywhere with any language.

But, in reality cases, why I need setup group of agents in different location? are some agent should close to somewhere? or some host could use some specific tools or could access to some third part system? could some one help to clarify it, thanks

**Comment by ekzhu** [comment, marked as answer]
There are several scenarios:

1. Agent implemented in different languages. E.g., .NET and Python agents
2. When the author of the agents (i.e. app dev) is different from the platform provider (i.e. cloud) that deploys and runs the agent.  Each agent is deployed into its own worker runtime inside a sandbox container, while all agents are connected to a central runtime host. Imagine you build an agent platform and let your users to create their own agents that can communicate with other agents, you can't trust your user's code to run directly inside your platform. 
3. Each agent might be implemented in its own stack and does not share dependencies with other agents.

Generally, scenarios that have agents from different systems talking to each other, you need to have distributed agents. 

It's still in the early days.

**Reply by vincent-pli** [reply]
Great, thanks. as you mentioned: **agents should belong to different systems**, that's the key point.
I guess maybe I was confused by `distributed`, make me start to think about `kubernetes` or `ray`. the case is totally different, right?

**Reply by MohMaz** [reply]
In this context, distributed means "agents running in different processes". These processes could run on one or multiple machines based on your design needs.
